### ISSUE TITLE
bug: program (in my case, brimstone) not having upgrade (from AI terminal)

### ISSUE THREAD

**Issue by kalecgos13** [Mobbstar/InvisibleInc-MoreMissions#96]
So I discovered a bug today with the AI terminal mission and the brimstone program.

As you can see in this picture, I upgrade Brimstone with a +1 firewall broken modifier.
![20220108184456_1](https://user-images.githubusercontent.com/38404765/148654657-a27e2271-a976-4947-8d62-c75ef6043a44.jpg).

However when I start the next mission, the modifier is gone and brimstone breaks 1 firewall (see picture).
![20220108185536_1](https://user-images.githubusercontent.com/38404765/148654702-3e738aed-4aeb-40e0-b484-9040d78ae10b.jpg)

This is the second time this happened, so I also have two save files backed up. One before starting the new mission and one during the new mission, just tell me if you need them and I will drop them here.

**Comment by Hekateras** [comment]
Sure, drop the saves but definitely also drop the log (one folder up from the saves, and then in Logs). Depending on how many mods you have installed, I might need your Mods folder, too, but let's look at the log first

**Comment by kalecgos13** [comment]
Here is the log [invisibleinc.txt](https://github.com/Mobbstar/InvisibleInc-MoreMissions/files/7834014/invisibleinc.txt)
And here are the two save files (it happened in the 4th save slot btw): [saves.zip](https://github.com/Mobbstar/InvisibleInc-MoreMissions/files/7834020/saves.zip)

idk if the log tells you much. If you need a better log, i can also just restart the day and redo the AI terminal level and then send you the log. As I havent finished the day yet and it seems to happen everytime.

**Comment by kalecgos13** [comment]
I got a better log file: [invisibleinc.txt](https://github.com/Mobbstar/InvisibleInc-MoreMissions/files/7835559/invisibleinc.txt).

This log file has me starting an AI terminal missions (then restarting it, because I only got given the option to alter the cooldown and range) and starting a new mission after that. 

The same bug occurs, but it seems to also happen with other modifiers, as this time I choose to reduce the PWR.

**Comment by Hekateras** [comment]
What was the mission you started after AI Terminal?

It looks like I will need your Mods folder. Can you upload it to something like Dropbox or Google Drive?
I need the same mods you have (same versions and all) to load the save and reproduce the bug.

**Comment by kalecgos13** [comment]
I triggered this bug around three times:
The first time I found it, I started another AI terminal (dont remember the corp) afterwards.
The second and third time, I started an FTM cybernetics lab.

And [here](https://www.dropbox.com/s/b3mz3d7yka9oh5e/mods.zip?dl=0) is a dropbox link to my whole mods folder.

**Comment by Hekateras** [comment]
Thanks for the files, this helps. Sort of.

I haven't been able to reproduce the bug - meaning I can load up your save, with your mods, do an AI Terminal, and any upgrades I make to Brimstone stick just fine.

Someone already reported this issue with Parasite once (no files to share though, just a vague description), but like now, I couldn't reproduce it. So this might be some kind of rare mod interaction. Do you tend to switch between different save slots/campaigns during the same game session (i.e. without restarting the game)? That can result in odd behaviour from mods.

I'd like to get to the bottom of this, so if it's not too much trouble, here's what I'd like you to do:
- Keep playing this campaign as normally (or, if you start a different campaign, keep the same mod settings. An easy way to ensure this is to retire campaign and click the 'restart campaign' button, this carries over your settings. Or get the Preset Saver mod.)
- Any time you do an AI Terminal, and do an upgrade, back up the save shortly before you exit the mission, then send it in again if next mission the upgrade isn't there. (Include your Mods folder again if you've hit Refresh Mods since then.)

One last, unrelated thing: The version of MM you're playing with (0.21.1) has a fatal bug that occurs at the end of the first midway DLC mission. It's patched in 0.21.2. Since you're about to hit Day 4 on that save slot, remember to update MM before then!

**Comment by Hekateras** [comment]
If by any chance you start a new campaign, remember you can always disable missions (even vanilla ones) on campaign start. So you could disable every mission type than AI Terminal. (I normally just debug spawn them in, but I can't really recommend that as a troubleshooting measure because that requires some familiarity with using debug mode.) Good luck!

**Comment by kalecgos13** [comment]
hmmm, I tried triggering the bug again. But I cannot seem to trigger it now.

I tried it also with debug mode, triggering the bug I found with New Items and Augments, rewinding mid upgrade etc.
But nothing worked. 

I did however find two other small things with the mission in the newest update, but I will make another issue for that.

**Comment by Hekateras** [comment]
May have a lead on the 'upgrade not carrying over' bug.

Do you remember if you used Retry Level in AI Terminal?

**Comment by Hekateras** [comment]
Fixed
https://github.com/Mobbstar/InvisibleInc-MoreMissions/pull/127
